week05/DeleteNodeInLinkedList.cpp

```diff
@@ -0,0 +1,9 @@
+// TC: O(1)
+// MC: O(1)
+class Solution {
+public:
+    void deleteNode(ListNode* node) {
+        node->val = node->next->val;
+        node->next = node->next->next;
```

Comment:
The node actually does not get deleted in this process. Try to `free` or `delete` the node.